sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -102,21 +88,23 @@ class RapidsHostMemoryStore(
       buffer: RapidsBuffer,
       catalog: RapidsBufferCatalog,
       stream: Cuda.Stream): Boolean = {
-    // this spillStore has a maximum size requirement (host only). We need to spill from it
-    // in order to make room for `buffer`.
-    val targetTotalSize = maxSize - buffer.memoryUsedBytes
-    if (targetTotalSize <= 0) {
-      // lets not spill to host when the buffer we are about
-      // to spill is larger than our limit
-      false
-    } else {
-      val amountSpilled = synchronousSpill(targetTotalSize, catalog, stream)
-      if (amountSpilled != 0) {
-        logDebug(s"Spilled $amountSpilled bytes from ${name} to make room for ${buffer.id}")
-        TrampolineUtil.incTaskMetricsDiskBytesSpilled(amountSpilled)
+    maxSize.forall { ms =>
+      // this spillStore has a maximum size requirement (host only). We need to spill from it
+      // in order to make room for `buffer`.
+      val targetTotalSize = ms - buffer.memoryUsedBytes
+      if (targetTotalSize <= 0) {
```

Comment:
Nit: Off-by-1 error (don't think this will matter in practice)
```suggestion
      if (targetTotalSize < 0) {
```